crates/worker/src/partition_processor_manager/mod.rs

```diff
@@ -702,6 +742,73 @@ impl PartitionProcessorManager {
         }
     }
 
+    fn spawn_create_snapshot_task(
+        &mut self,
+        partition_id: PartitionId,
+        result_sender: oneshot::Sender<anyhow::Result<SnapshotId>>,
+        archived_lsn_sender: watch::Sender<Option<Lsn>>,
+    ) {
+        debug!("In spawn create snapshot task...");
+
+        if self
+            .pending_snapshot_export_tasks
+            .get(&partition_id)
+            .is_some_and(|task| !task.is_finished())
+        {
+            warn!(%partition_id, "Snapshot creation already in progress, rejecting request");
+            result_sender
+                .send(Err(anyhow!("Snapshot creation already in progress")))
+                .ok();
+        } else {
+            let config = self.updateable_config.live_load();
+
+            let snapshot_base_path = config.worker.snapshots.snapshots_dir(partition_id);
+
+            let create_snapshot_task = SnapshotPartitionTask {
+                cluster_name: config.common.cluster_name().into(),
+                node_name: config.common.node_name().into(),
+                partition_id,
+                partition_store_manager: self.partition_store_manager.clone(),
+                snapshot_base_path,
+                result_sender,
+                archived_lsn_sender,
+            };
+            let snapshot_span = tracing::info_span!("create-snapshot");
+
+            let spawn_task_result = restate_core::task_center().spawn_unmanaged(
+                TaskKind::PartitionSnapshotProducer,
+                "create-snapshot",
+                Some(partition_id),
+                async move {
+                    create_snapshot_task
+                        .create_snapshot()
+                        .await
+                        .inspect_err(|err| {
+                            warn!("Unhandled error in create_snapshot task: {}", err)
+                        })
+                }
+                .instrument(snapshot_span),
+            );
```

Comment:
Is it safe to spawn the snapshot tasks on the Tokio runtime? What we should avoid is if any blocking operations block the Tokio threads so that it would interfere with the normal processing.